Formula/php.rb

```diff
@@ -0,0 +1,383 @@
+      #{pear_prefix}/.lock
+    ]
+
+    %W[
+      #{pear_prefix}/.channels
+      #{pear_prefix}/.channels/.alias
+    ].each do |f|
+      chmod 0755, f
+      pear_files.concat(Dir["#{f}/*"])
+    end
+
+    chmod 0644, pear_files
+
+    # Custom location for extensions installed via pecl
+    pecl_path = HOMEBREW_PREFIX/"lib/php/pecl"
+    ln_s pecl_path, prefix/"pecl" unless (prefix/"pecl").exist?
+    php_basename = File.basename Utils.popen_read("#{bin}/php-config --extension-dir").chomp
+    php_ext_dir = opt_prefix/"lib/php"/php_basename
+
+    # fix pear config to install outside cellar
+    pear_path = HOMEBREW_PREFIX/"share/pear"
+    {
+      "php_ini" => etc/"php/#{php_version}/php.ini",
+      "php_dir" => pear_path,
+      "doc_dir" => pear_path/"doc",
+      "ext_dir" => pecl_path/php_basename,
+      "bin_dir" => opt_bin,
+      "data_dir" => pear_path/"data",
+      "cfg_dir" => pear_path/"cfg",
+      "www_dir" => pear_path/"htdocs",
+      "man_dir" => HOMEBREW_PREFIX/"share/man",
+      "test_dir" => pear_path/"test",
+      "php_bin" => opt_bin/"php",
+    }.each do |key, value|
+      value.mkpath if key =~ /(?<!bin|man)_dir$/
+      system bin/"pear", "config-set", key, value, "system"
+    end
+
+    %w[
+      ldap
+      imap
+      opcache
+      snmp
+      zip
+    ].each do |e|
+      ext_config_path = etc/"php/#{php_version}/conf.d/ext-#{e}.ini"
+      extension_type = (e == "opcache") ? "zend_extension" : "extension"
+      if ext_config_path.exist?
+        inreplace ext_config_path,
+          /#{extension_type}=.*$/, "#{extension_type}=#{php_ext_dir}/#{e}.so"
+      else
+        # Do not load snmp extension by default
+        extension_type.prepend(";") if e == "snmp"
+        ext_config_path.write <<~EOS
+          [#{e}]
+          #{extension_type}="#{php_ext_dir}/#{e}.so"
+        EOS
+      end
+    end
+  end
+
+  def php_version
+    version.to_s.split(".")[0..1].join(".")
+  end
+
+  plist_options :startup => true, :manual => "php-fpm"
+
+  def plist; <<~EOS
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>KeepAlive</key>
+        <true/>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>ProgramArguments</key>
+        <array>
+          <string>#{opt_sbin}/php-fpm</string>
+          <string>--nodaemonize</string>
+        </array>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>WorkingDirectory</key>
+        <string>#{var}</string>
+        <key>StandardErrorPath</key>
+        <string>#{var}/log/php-fpm.log</string>
+      </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    system "#{bin}/php", "-i"
+    system "#{sbin}/php-fpm", "-t"
+    system "#{bin}/phpdbg", "-V"
+    system "#{bin}/php-cgi", "-m"
+    shell_output("#{bin}/php -r '(extension_loaded(\"Zend OPCache\"))?exit(0):exit(1);'")
```

Comment:
This test seems a bit redundant. You could just as easily look for the extension in the output produced in the first `php -i` test.